### PR TITLE
Implement client certificate verification

### PR DESCRIPTION
**Pull request by NteRySin**
Improve SSL library handling as per https://docs.python.org/3/library/ssl.html#server-side-operation and https://docs.python.org/3/library/ssl.html#best-defaults
Implement client certificate verification as per https://docs.python.org/3/library/ssl.html#verifying-certificates

### CODE REVIEW

**Review by Densaugeo — COMMENTED** [review]
Neat! I didn't know about mutual TLS.

I have a couple review comments on the code. Additionally:
- Please add an example to the readme similar to the HTTPS example, showing how to create whichevever certificates are necessary, start a server and upload a file.
- We'll need test cases. If setting up the certificates for mTLS isn't too complicated, I can set those up.

File: uploadserver/__main__.py [hunk 1]
```diff
@@ -12,18 +12,33 @@ def ssl_wrap(socket):
...
+        
+        if args.client_certificate:
+            context.load_verify_locations(cafile=args.client_certificate)
+            context.verify_mode = ssl.CERT_REQUIRED
+
+        return context.wrap_socket(socket, server_side=True)
     except FileNotFoundError:
-        print('Certificate \'{}\' not found, exiting'.format(args.certificate))
+        if args.client_certificate:
+            print('Server certificate \'{}\' or client certificate \'{}\' not found, exiting'.format(args.server_certificate, args.client_certificate))
```
[the remark this answers is not in the document]
Comment (NteRySin) [reply uploadserver/__main__.py]: Thanks for pointing it out, I improved it.

File: uploadserver/__main__.py [hunk 2]
```diff
@@ -46,12 +61,12 @@ def test(HandlerClass=BaseHTTPRequestHandler,
...
         with ServerClass(server_address, HandlerClass) as httpd:
-            if args.certificate: httpd.socket = ssl_wrap(httpd.socket)
+            if args.server_certificate: httpd.socket = ssl_wrap(httpd.socket)
             
             sa = httpd.socket.getsockname()
-            serve_message = "Serving {proto} on {host} port {port} ({proto_lower}://{host}:{port}/) ..."
+            serve_message = "Serving {proto} on {host} port {port} ({proto_lower}://{host}:{port}/) {mTLS}..."
             print(serve_message.format(host=sa[0], port=sa[1], proto=uploadserver.PROTOCOL, 
-                proto_lower=uploadserver.PROTOCOL.lower()))
+                proto_lower=uploadserver.PROTOCOL.lower()), mTLS=("with mutual TLS " if args.client_certificate else ""))
```
[the remark this answers is not in the document]
Comment (NteRySin) [reply uploadserver/__main__.py]: Ok, I removed it.

**Comment by NteRySin** [comment]
I also updated the documentation, it should give a guidance for using client authentication.

**Comment by Densaugeo** [comment]
Excellent!

One last request before I merge this: please change the client_certificate and server_certificate CLI options to client-certificate and server-certificate, for easier typing and consistency with cURL's naming convention.

**Pull request merged**